### DISCUSSION TITLE
How to contribute manual

### DISCUSSION THREAD

**Discussion by ensismoebius** [focustimerhq/FocusTimer#634] [General]
There is some guidelines on how to contribute to project? Which tools should we use?

**Comment by kamilprusko** [comment]
Hi André :)

There is a draft :D Please leave feedback if you struggle with something.
https://github.com/gnome-pomodoro/gnome-pomodoro-website/tree/wip/mkdocs/docs/docs/get-involved

In short, we try to follow guidelines made by GNOME:
https://wiki.gnome.org/Projects/GnomeShell/StyleGuide

You should be comfortable installing the app using meson. I use gedit to edit code and reinstall the app every time I make changes. For long time I resorted to gedit because it's hard to find a good IDE supporting Vala or gjs. You could try GNOME Builder.

To test the extension, it's best to run gnome-shell from terminal
```
dbus-run-session -- gnome-shell --nested --wayland
```

It's safer; easier to monitor gnome-shell logs. GNOME Shell doesn't properly reload extensions which makes it mandatory.

I use jhbuild with latest gnome-shell from git. But it's not necessary.
https://wiki.gnome.org/HowDoI/Jhbuild

Handy documentation:
https://gjs-docs.gnome.org/
https://valadoc.org/glib-2.0/index.html
https://wiki.gnome.org/Projects/Vala/Documentation
https://gjs.guide/extensions/development/debugging.html#restarting-gnome-shell

I currently work on porting the app to GTK 4 https://github.com/gnome-pomodoro/gnome-pomodoro/tree/wip/gtk4 .
In wip/gtk4 branch it's easier to start development using GNOME Builder & flatpak. But the app there is nowhere at working state yet. FYI, I intend to remove the plugin architecture based on Peas, it wouldn't be much useful when we distribute the app as flatpak.

What things you would like to work on?

Best regards

**Reply by ensismoebius** [reply]
Thank you!
